stats/opencensus/stats.go

```diff
@@ -86,26 +87,36 @@ func (csh *clientStatsHandler) statsTagRPC(ctx context.Context, info *stats.RPCT
 
 	// Populate gRPC Metadata with OpenCensus tag map if set by application.
 	if tm := tag.FromContext(ctx); tm != nil {
-		ctx = stats.SetTags(ctx, tag.Encode(tm))
+		md, ok := metadata.FromOutgoingContext(ctx)
```

Comment:
`metadata.AppendToOutgoingContext` is more efficient.  And as long as we read the last incoming instance of the header, I believe the behavior is similar enough.  And if anyone is setting the header multiple times to different values, they were just lucky if it ever worked.

Reply:
Changed